### PR TITLE
Fix for database tables not being created in Heroku

### LINKED ISSUE #42: Database tables are not being created on Heroku
When running the project locally, tables seems to be created as expected, but on Heroku deploy it is not (raising a exception with the message "no table 'users'").

Tasklist:
- [ ] Ensure that locally, in a clean environment (no database file or other generated content), if database is being created.
  - [ ] If it is not, try to fix it locally (always ensuring a clean environment).
- [ ] Make sure (still locally) that a previous content isn't overwritten when restarting the server.
- [ ] Deploy to heroku and repeat the steps that were done locally ensuring Heroku doesn't get configured to delete the previous database file.

### PR DESCRIPTION
**Pull request by erik-whiting**
# Description
## Database Tables not Being Created
In the `codecompl.py` file, I added import `db` to the import statement. With that models in the namespace, we can then call `db.create_all()` and `db.session.commit()` which will create the database tables. These functions don't wipe out any data locally; as long as there are database files (the ones ending with `.db`) the data still persists.

## Deploying to Heroku
In order to deploy this to my own Heroku instance, I had to use Pipfile. I could not get this to deploy with poetry in my own Heroku instance, but it's working in the actual environment according to JPTIZ, so I left out my changes related to that.

## Quick note on SQLite
According to [this article](https://devcenter.heroku.com/articles/dynos#ephemeral-filesystem) from Heroku themselves:
>If you were to use SQLite on Heroku, you would lose your entire database at least once every 24 hours.

However, JPTIZ says the data is persisting in the Heroku instance so I'm probably doing something wrong with Heroku

Fixes #42 
 
## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?
## Local Test - No Database
1. Make code changes
2. Go to journals page, see that there is no error
3. Use postman to create a new journal record
4. Go to journals page, see that there is no error and that the record I created exists

## Local Test - Existing Database
1. After completing "*Local Test - No Database*", stop the server
2. Restart the server
3. Go to journals page, see that there is no error and that the record still exists

## Local Test - Do it All Over
1. I deleted the database files
2. Repeated both test cases as above
3. Saw that everything still works as expected

**Test Configuration**:

* Python version: 3.9.4
* Poetry version: 1.1.13

# Checklist:

- [X] I have readed the [Contributing Guidelines](https://github.com/codeisscience/journal-policy-tracker-backend/blob/master/CONTRIBUTING.md).

- [X] My behaviour/Contribution abides by the [Code of Conduct](https://github.com/codeisscience/journal-policy-tracker-backend/blob/master/CODE_OF_CONDUCT.md).

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- _**N/A**_
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- I manually tested my changes but there are no existing tests in this repository so I wouldn't know where to put them
- [ ] New and existing unit tests pass locally with my changes
- _**N/A**_

### CODE REVIEW

**Review by JPTIZ — CHANGES_REQUESTED** [review]
Since we already use [Poetry](https://python-poetry.org/) (see this repo's [pyproject.toml](https://github.com/codeisscience/journal-policy-tracker-backend/blob/master/pyproject.toml)), I wouldn't add Pipenv files unless, for some reason, it is strictly necessary.

About the changes, as much as I tested (first from your branch after deleting journals.db, then  from master deleting journals.db beforehand as well) it works :) So removing the Pipenv files and addressing my other comment should be enough for approval.

(Oh, and thank you for fixing this!)

File: codecompl.py [hunk 1]
```diff
@@ -1,4 +1,9 @@
-from server import app
+from server import app, db
+from server.models import *
```
Comment (JPTIZ) [inline codecompl.py:2]:
```suggestion
```

**Comment by erik-whiting** [comment]
Hey @JPTIZ , thanks so much for the feedback.

I wanted to mention one thing regarding the use of `*` vs explicitly importing classes. This `import` statement is the part that builds the tables upon deployment. Say we add a new table called `Foo`; we wouldn't have to make any changes to this file in order to build the `Foo` table upon deployment (provided we define the class in the right place). With explicit importation, we'll have to come back to this file any time we add a table to the database (e.g., `from db.models import User, Journal` would have to become `from db.models import User, Journal, Foo`).

I'll definitely make the requested changes, just be advised that when a new table is added, it'll have to be explicitly imported in this file as well.

Regarding using poetry to deploy, I could not get my own Heroku instance of this application to deploy correctly without using a pipfile. I'll go ahead and remove that in this PR, but I won't be able to test it on my own instance, you may have to double check everything deploys ok yourself 😬 

I'll tag you again once the changes are made, thanks again so much for getting back to me!

**Comment by erik-whiting** [comment]
@JPTIZ thanks again for your feedback yesterday, I've made the requested changes. Let me know if I can help with anything else :)

**Review by JPTIZ — APPROVED** [review]

**Pull request merged**